flake.nix

```
@@ -8,22 +8,40 @@
     };
 
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
-
+    naersk.url = "github:nmattia/naersk";
     utils.url = "github:numtide/flake-utils";
   };
 
-  outputs = { self, flake-compat, nixpkgs, utils }: utils.lib.eachDefaultSystem (system: let
+  outputs = { self, flake-compat, nixpkgs, utils, naersk }: utils.lib.eachDefaultSystem (system: let
     pkgs = import nixpkgs { inherit system; };
-  in {
-    packages.nixpkgs-hammer = pkgs.runCommand "nixpkgs-hammer" {
+    naersk-lib = naersk.lib."${system}";
+  in rec {
+
+    packages.ast-checks = naersk-lib.buildPackage {
+      name = "ast-checks";
+      root = ./ast-checks;
+    };
+
+    packages.nixpkgs-hammer = let
+      # Find all of the binaries installed by ast-checks. Note, if this changes
+      # in the future to use wrappers or something else that pollute the bin/
+      # directory, this logic will have to grow
+      ast-check-names = let
+        binContents = builtins.readDir "${packages.ast-checks}/bin";
+      in
+        pkgs.lib.mapAttrsToList (name: type: assert type == "regular"; name) binContents;
```

It does look a bit cleaner:

```suggestion
      # Find all of the binaries installed by ast-checks.
      ast-check-names = let
        cargo = builtins.fromTOML (builtins.readFile ./ast-checks/Cargo.toml);
      in
        map (bin: bin.name) cargo.bin;
```

---

Applied